pkg/controllers/daemonset/controller.go

```diff
@@ -0,0 +1,89 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package daemonset
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/karpenter-core/pkg/controllers/state"
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+)
+
+// Controller for the resource
+type Controller struct {
+	kubeClient client.Client
+	cluster    *state.Cluster
+}
+
+// NewController constructs a controller instance
+func NewController(kubeClient client.Client, cluster *state.Cluster) corecontroller.Controller {
+	return &Controller{
+		kubeClient: kubeClient,
+		cluster:    cluster,
+	}
+}
+
+func (c *Controller) Name() string {
+	return "daemonset"
+}
+
+// Reconcile the resource
+func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	pods := &v1.PodList{}
+	var daemonSet appsv1.DaemonSet
+	err := c.kubeClient.List(ctx, pods)
```

Comment:
```suggestion
	err := c.kubeClient.List(ctx, pods, client.InNamespace(daemonSet.Namespace))
```

We can definitely limit list by the namespace that the daemonset is in